### PR TITLE
Fixes a few runtimes

### PR DESCRIPTION
**Pull request by gbasood**
RUNTIME
FIXES

Someone is really fond of using src by the way
I really do hope everyone understands that src refers to thing that owns the proc being executed?

### CODE REVIEW

**Comment by gbasood** [comment]
On all the logic checks I added to vamp code: if there's no mind and no mind.current, then either the vamp is gone or his body is gone, in either case we shouldn't be doing any more logic, so it's **FINE**

All of those blocks would runtime if either mind or mind.current broke though, so they're useful.

**Comment by Intigracy** [comment]
but ANTAG REFACTOR

**Comment by PJB3005** [comment]
If I ever finish that.

File: code/modules/hydroponics/seed_mobs.dm [hunk 1]
```diff
@@ -57,6 +57,9 @@
 		O = pick(currently_querying)
 		while(currently_querying.len && !check_observer(O)) //While we the list has something and
 			currently_querying -= O				//Remove them from the list if they don't get checked properly
+			if(currently_querying.len) // Are we out of people?
```
Comment (clusterfack) [inline code/modules/hydroponics/seed_mobs.dm]: I think this is meant to be a repeating loop for querying users to become a diona, this break if it has no length will only make it loop once effectively not making it loop at all.

**Comment by clusterfack** [comment]
Otherwise looks good

File: code/modules/hydroponics/seed_mobs.dm (hunk 1 above)
Comment (gbasood) [inline code/modules/hydroponics/seed_mobs.dm:60]: The `break` breaks the while loop, which checks for a thing in currently_querying and while O isn't an observer
currently_querying is never added to inside the while block
There is no sleep in the while block for it to wait for new input
`O` is only assigned outside the while loop and within it, if the last O wasn't an observer. And it's de-referrenced only if O is picked (with this PR)

The way this loop normally functions is to check if anyone in currently_querying isn't an observer, and then remove them. That's it.

**Pull request merged**